src/cmd-buildupload

```diff
@@ -22,12 +22,16 @@ CACHE_MAX_AGE_ARTIFACT = 60 * 60 * 24 * 365
 # set metadata caching to 5m
 CACHE_MAX_AGE_METADATA = 60 * 5
 from cosalib.builds import Builds, BUILDFILES
-from cosalib.cmdlib import load_json, retry_stop, retry_boto_exception, retry_callback  # noqa: E402
+from cosalib.cmdlib import load_json, retry_stop, retry_boto_exception, retry_callback, write_json  # noqa: E402
+
 
 
 def main():
     args = parse_args()
-    args.func(args)
+    if args.record_s3_metadata:
```

Comment:
Hmm, so is the idea here that the original `meta.json` in the S3 bucket remains unchanged, and only the one that we upload to Brew gets changed?

I think that sounds good, but the interface here seems a bit confusing. Maybe this should instead be part of `cosa koji-upload`? E.g. `cosa koji-upload --s3 my-bucket/my/path/to/this/meta.json` ?

Reply:
We could make it a regular switch to `cosa buildupload s3` that adds the S3 information that's already provided anyway into the `meta.json`. But I think that could lead to confusion (e.g. what should we do if we `buildupload` the same build multiple times to different buckets?).

Reply:
> Hmm, so is the idea here that the original meta.json in the S3 bucket remains unchanged, and only the one that we upload to Brew gets changed?

Yeah, I was thinking of that, you can change the meta.json in a separate phase. But maybe we can update it before sending to S3

 >Maybe this should instead be part of cosa koji-upload? E.g. cosa koji-upload --s3 my-> bucket/my/path/to/this/meta.json ?
 
 umm, I think  koji-upload does a different action, we would be mixing koji(brew) with S3. But I'm not against it.


> (e.g. what should we do if we buildupload the same build multiple times to different buckets?).

I think we only care about the data not the locations, the Prodsec needs to know a place where it is. At least it is my understanding.

Reply:
>> Maybe this should instead be part of cosa koji-upload? E.g. cosa koji-upload --s3 my-> bucket/my/path/to/this/meta.json ?
>
> umm, I think koji-upload does a different action, we would be mixing koji(brew) with S3. But I'm not against it.

To clarify, I don't mean that `koji-upload` would upload to S3. Only it would add the S3 information to the `meta.json` that gets uploaded to Brew.

Reply:
@jlebon Changed, let me know if that makes sense for you.